### PR TITLE
iOS Background Task for Exposure Detection

### PR DESCRIPTION
**Pull request by mattThousand**
### Why
We'd like `Exposure Detection` to be carried out at regular intervals in the background, so that the user remains current on their possible exposures.

### This Commit
This commit registers and schedules a background task to run on iOS that periodically executes exposure detection in the background. It also implements a local notification to encourage the user to enable bluetooth if exposure notifications have been authorized but bluetooth has not be enabled.

** Note: the `BT` app does not currently request notification permissions. This is something we should request for the `BT` app before ship in order for notifications to appear for the user

Jun-19-2020 17-18-20

### CODE REVIEW

File: ios/BT/ExposureManager.swift [hunk 1]
```diff
@@ -2,245 +2,228 @@ import Foundation
...
+  
+  enum EnabledState: String {
+    case ENABLED, DISABLED
+  }
+  
+  enum AuthorizationState: String {
+    case AUTHORIZED, UNAUTHORIZED
+  }
+  
+  typealias DeviceState = (AuthorizationState, EnabledState)
```
Comment (johnschoeman) [inline ios/BT/ExposureManager.swift:26]: 💯

File: ios/BT/ExposureManager.swift [hunk 2]
```diff
@@ -2,245 +2,228 @@ import Foundation
...
+    do {
+      try BGTaskScheduler.shared.submit(taskRequest)
+    } catch {
+      print("Unable to schedule background task: \(error)")
+    }
+  }
+
+  func notifyUserBlueToothOffIfNeeded() {
+    let identifier = String.bluetoothNotificationIdentifier
+    if ENManager.authorizationStatus == .authorized && manager.exposureNotificationStatus == .bluetoothOff {
```
Comment (johnschoeman) [inline ios/BT/ExposureManager.swift:239]: so does this mean that if the user has bluetooth off, but en permissions on we dont have en notifications coming through?  or does turning bluetooth off also turn en permission off automatically?

**Review by johnschoeman — APPROVED** [review]
LGTM! a question.

It was also a bit hard to distinguish between the diff that was due to the files being moved around and the actual new code.  If you think its useful, it might be worth pairing on this review!

**Comment by johnschoeman** [comment]
and a card has been cut for the push notification work: https://trello.com/c/UyWrnN0E/66-as-a-user-when-i-am-onboarding-the-bt-app-i-am-asked-to-provide-push-notification-permissions

File: ios/BT/ExposureManager.swift (hunk 2 above)
Comment (mattThousand) [reply ios/BT/ExposureManager.swift:239]: good question - the latter; I'll add a comment 👍

**Pull request merged**